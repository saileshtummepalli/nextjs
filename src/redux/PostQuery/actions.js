export const postActionTypes = {
  FAILURE: "FAILURE",
  CREATE_DATA: "CREATE_DATA",
  CREATE_DATA_SUCCESS: "CREATE_DATA_SUCCESS",
  HYDRATE: "HYDRATE",
};

export function failure(error) {
  return {
    type: postActionTypes.FAILURE,
    error,
  };
}

export const CreateData = ({ data }) => {
  return {
    type: postActionTypes.CREATE_DATA,
    data,
  };
};

export const createDataSaga = ({ data }) => {
  return {
    type: postActionTypes.CREATE_DATA_SUCCESS,
    data,
  };
};
