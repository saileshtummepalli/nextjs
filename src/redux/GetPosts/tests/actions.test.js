import axios from "axios";
import configureMockStore from "redux-mock-store";
import  {loadDataSuccess } from "../actions";
 
import loadDataSaga from "../saga"

const mockStore = configureMockStore();
const initialState = {}
const store = mockStore(initialState)
describe("getPosts loadDataSucess", () => {
    const data = {
      name: "Leanne Graham",
      email: "Sincere@april.biz",
      phone: "1-770-736-8031 x56442",
      steet: "Kulas Light",
      city: "Gwenborough",
    };
    it('action', () => {
      store.dispatch(loadDataSuccess(data));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
