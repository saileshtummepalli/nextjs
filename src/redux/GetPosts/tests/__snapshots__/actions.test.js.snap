// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPosts loadDataSucess action 1`] = `
Array [
  Object {
    "data": Object {
      "city": "Gwenborough",
      "email": "Sincere@april.biz",
      "name": "Leanne Graham",
      "phone": "1-770-736-8031 x56442",
      "steet": "Kulas Light",
    },
    "type": "LOAD_DATA_SUCCESS",
  },
]
`;
